src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -164,6 +221,42 @@ def __init__(self, parent=None):
         self.lblUnity.setStyleSheet(new_font)
         self.lblUnitz.setStyleSheet(new_font)
 
+    def eventFilter(self, target, event):
+        """A function to catch FocusOut on QLineEdit widgets in the GUI
+
+        This function checks whether the text in a textbox is valid when the user clicks out of it
+        or in some other way loses focus to the widget (e.g. TAB, pressing the compute button). If the
+        text contained is not accpetable to the validator (intermediate state, e.g. empty) it restores
+        the previous last accepted value.
+
+
+        :param target: The widget which lost focus and triggered the eventFilter
+        :type target: QObject
+        :param event: the event which triggered the eventFilter
+        :type event: QEvent
+        """
+        if target in self.lineEdits and event.type() == QtCore.QEvent.FocusOut:
+            if target.hasAcceptableInput():
+                self.last_acceptable_values[target] = target.text()
```

Comment:
As of 86a0d50218da0ef44e445b82f4db43af8ac849b6 I have altered the process so that much like with file validation if the textbox has an invalid value (what Qt considers an 'intermediate' state in the regex) it is highlighted red and functionality is disabled via the same method as file validation. In keeping with prior functionality (the warnings on the no. of Q bins and Qx max.) this process happens as the user types, rather than when the textbox loses focus. There are now two different highlighted states:

 - Errors reflect 'intermediate' regex values and cannot be used to draw/save/compute data. The textbox is highlighted red and functionality is disabled. These are the values that were previously coerced
 - Warnings reflect values which may produce invalid results, such as numerical artefacts. These are the values which were previously highlighted red. These are now highlighted orange and as before do not disable any functionality.